templates/components/vectordbs/python/llamacloud/index.py

```diff
@@ -82,14 +84,63 @@ def to_index_kwargs(self) -> dict:
         }
 
 
-def get_index(config: IndexConfig = None):
+def get_index(
+    config: IndexConfig = None,
+    create_if_missing: bool = False,
+):
     if config is None:
         config = IndexConfig()
-    index = LlamaCloudIndex(**config.to_index_kwargs())
-
-    return index
+    # Check whether the index exists
+    try:
+        index = LlamaCloudIndex(**config.to_index_kwargs())
+        return index
+    except ValueError:
+        logger.warning("Index not found")
+        if create_if_missing:
+            logger.info("Creating index")
+            _create_index(config)
```

Comment:
We don't need the ensure function because it's now resolved by the framework. Instead, i moved the create index code to index file and call it here